src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private Element<T> topOfStack;
+    private int size;
+
+    private static class Element<T> {
+        private Element<T> prev;
+        private T value;
+
+        public Element(Element<T> prev, T value) {
+            this.prev = prev;
+            this.value = value;
+        }
+    }
+
     public void push(T value) {
 
+        if (size == 0) {
+            topOfStack = new Element<>(null, value);
+        } else {
+            Element<T> oldTopValue = topOfStack;
+            topOfStack = new Element<>(oldTopValue, value);
+        }
+        size++;
     }
 
     public T peek() {
-        return null;
+        sizeCheck();
+        return topOfStack.value;
+
     }
 
     public T pop() {
-        return null;
+        sizeCheck();
```

Comment:
The sizeCheck method is correctly called before popping an element to ensure the stack is not empty.